src/Config.ts

```diff
@@ -1,32 +1,84 @@
-import { open } from "node:fs/promises";
+import { open, access, constants, FileHandle } from "node:fs/promises";
 import Logger from "./Logger.js";
 
-export default class Config {
-    public port: number = 25565;
-    public logLevel: Logger.Level = Logger.Level.INFO;
+
+export interface Config {
+    /**
+     * Port to listen on 
+    */
+    port: number;
+
+    /**
+     * The level to display logs at
+     */
+    logLevel: Logger.Level;
 
     /**
      * Kick reason for when the server is shutting down
      */
-    public shutdownKickReason: string = "Server closed";
+    shutdownKickReason: ChatComponent;
+}
 
+export class ConfigLoader {
     /**
      * Get a Config instance from a json file
      * @param file The file to read from
      * @returns a promise that resolves to a Config instance
+     * @throws if the file does not exist or is invalid
```

Comment:
```suggestion
     * @throws {Error & {CODE: "EACCESS"}} failed to read config
     * @throws {SyntaxError} failed to parse config
```
since we don't have typed catch in TS, this is the closest we can get to knowing what errors to expect.

And bruh `fs` just uses `Error` with some added stuff (quite the contrast from java xd)

Reply:
typed error are so nice like
```js
catch (error: RuntimeException | ParseError)
``` and such xd